jax/_src/api.py

```diff
@@ -2814,6 +2814,25 @@ def eval_shape(fun, *args, **kwargs):
   (2000, 1000)
   >>> print(out.dtype)
   float32
+
+  In case the function has static arguments, it should be wrapped
+  using :func:`functools.partial` before passing to :func:`eval_shape`:
```

Comment:
This is a bit too strong: `partial` is one way to do this, you could also use a `lambda` or otherwise define a closure function. I might phrase it this way:
```
All arguments passed via :func:`eval_shape` will be treated as dynamic; static arguments can be included via closure, for example using :func:`functools.partial`:
```

Reply:
Agreed, I changed to your proposed sentence.